TouchPortal-API/TouchPortalAPI/__init__.py

```diff
@@ -246,8 +250,14 @@ def settingUpdate(self, settingName:str, settingValue):
             self.currentSettings[settingName] = settingValue
 
     def stateUpdate(self, stateId:str, stateValue:str):
-        if stateId and stateId not in self.currentStates or self.currentStates[stateId] != stateValue:
-            self.send({"type": "stateUpdate", "id": stateId, "value": stateValue})
+        self.__stateUpdate(stateId, stateValue, False)
+        
+    def __stateUpdate(self, stateId:str, stateValue:str, forced:bool):
```

Comment:
Any reason not to just add `force` to the public `stateUpdate()` method (with `False` default)? I was actually thinking this may be useful to have, but didn't have an actual use case.

Reply:
Because i don't see any use case and also think this shouldn't be used by plugins. Why would they need to force update?

Reply:
Because otherwise there's no way to override the automatic state cache?  Which perhaps is a bigger issue in general.

Apparently you found this useful for something...  :-)  I don't understand the use case for forcing a state update on _every_ page change either, but maybe someone may want to do something similar in other circumstances (and they can already iterate over the stored states if they want).  And I don't really see why the extra option would hurt (w/out "forcing" someone to use the "private" method which would exist anyway).

I'm not attached either way, but perhaps you could elaborate on why you think "this shouldn't be used by plugins."

Reply:
So to explain: In the [Touch Portal discord server](https://discord.gg/MgxQb8r) we had multiple complaints about states and images not showing up correctly after a page change. This is fixed by sending a state update. The developers of Touch Portal added this event for plugin devs to send a state update on page change. This is why i'm adding this.

For the "this shouldn't be used by plugins.":
I think plugin states shouldn't need to be updated if they stay the same. This is also why i added the state value cache.

Reply:
Maybe its up to @KillerBOSS2019 if he wants to have this parameter public

Reply:
OK, thanks for the explanation, and good to know. I have not noticed this at all with my plugin/TP, nor seen the Discord complaints.  

To me it sounds a bit crazy if a user has a bunch of plugins (or a few with a lot of states) and they all re-send all their states on every page change (even if the page being used has nothing to do with a plugin).  But I guess if that's a workaround to their refresh issue, so be it. Still not sure it should be on by default, but either way.

Maybe TP devs could add a way to determine if the page being switched-to is using a given plugin at all.  Or a way to request a state update from a plugin explicitly.   ... dreaming...   :)